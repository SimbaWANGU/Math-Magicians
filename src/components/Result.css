.resultContainer {
  background-color: lightgray;
  height: 20%;
  border-bottom: 1px solid black;
}

.resultContainer > p {
  font-size: 40px;
  width: 10%;
  float: right;
  height: 28%;
  margin: auto;
  margin-top: 5%;
  color: white;
}
