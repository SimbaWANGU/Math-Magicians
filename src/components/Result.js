/* eslint-disable react/destructuring-assignment */
import React from 'react';
import './Result.css';
import PropTypes from 'prop-types';

// eslint-disable-next-line react/prefer-stateless-function
class Result extends React.Component {
  render() {
    return (
      <input
        className="resultContainer"
        value={this.props.result}
        type="text"
        onKeyDown={(e) => e.preventDefault()}
        disabled
      />
    );
  }
}

Result.propTypes = {
  result: PropTypes.number,
};

Result.defaultProps = {
  result: 0,
};

export default Result;
