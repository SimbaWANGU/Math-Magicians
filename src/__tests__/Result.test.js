import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Result from '../components/Result';

describe('Result display tests', () => {
  const result = 0.1839870354127373;
  it('display properly the correct result', () => {
    const component = renderer.create(
      <Result result={result} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('the display should be a disabled input', () => {
    const { getByTestId } = render(<Result result={result} />);
    expect(getByTestId('resultDisplay')).toBeDisabled();
  });
});
