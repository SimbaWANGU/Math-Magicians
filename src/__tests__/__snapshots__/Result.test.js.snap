// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result display tests display properly the correct result 1`] = `
<input
  className="resultContainer"
  data-testid="resultDisplay"
  disabled={true}
  onKeyDown={[Function]}
  type="text"
  value={0.1839870354127373}
/>
`;
